dask_glm/families.py

```diff
@@ -1,18 +1,48 @@
 from __future__ import absolute_import, division, print_function
 
-from dask_glm.utils import dot, exp, log1p, sigmoid
+from dask_glm.utils import dot, exp, log1p, sigmoid, RegistryClass
 
 
-class Logistic(object):
+class Family(RegistryClass):
+    """Base class methods for distribution.
+
+    This class represents the required methods to add a new distribution to work
+    with the algorithms.
+    """
+
+    def loglikelihood(self, Xbeta, y):
+        raise NotImplementedError
+
+    def pointwise_loss(self, beta, X, y):
+        """Loss, evaluated point-wise."""
+        beta, y = beta.ravel(), y.ravel()
+        Xbeta = X.dot(beta)
+        return self.loglikelihood(Xbeta, y)
+
+    def pointwise_gradient(self, beta, X, y):
+        """Loss, evaluated point-wise."""
+        beta, y = beta.ravel(), y.ravel()
+        Xbeta = X.dot(beta)
+        return self.gradient(Xbeta, X, y)
+
+    def gradient(self, Xbeta, x, y):
+        raise NotImplementedError
+
+    def hessian(self, Xbeta, x):
+        raise NotImplementedError
+
+
+class Logistic(Family):
     """Implements methods for `Logistic regression`_,
     useful for classifying binary outcomes.
 
     .. _Logistic regression: https://en.wikipedia.org/wiki/Logistic_regression
     """
-    @staticmethod
-    def loglike(Xbeta, y):
+    name = 'logistic'
+
+    def loglikelihood(self, Xbeta, y):
         """
-        Evaluate the logistic loglikeliehood
+        Evaluate the logistic loglikelihoodlihood
```

Comment:
spelling.  Also, this led to some confusion for the Poisson family so could you call out that this is the *negative* log-likelihood for all the families?